### PR TITLE
Add start_record interface

### LINKED ISSUE #3127: add paddle_start_get_records interface in python interface

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #3127

### CODE REVIEW

File: go/pserver/client/c/test/test_train.py [hunk 1]
```diff
@@ -54,6 +54,9 @@ def main():
 
     # event_handler to print training and testing info
     def event_handler(event):
+        if isinstance(event, paddle.event.BeginPass):
+            master_client.paddle_start_get_records(event.pass_id)
```
Comment (helinwang) [inline go/pserver/client/c/test/test_train.py]: I think `paddle_start_get_records` is a implementation detail of cloud reader, we need to hide it inside the reader implementation.
The new pass is also indicated by `reader()` being called again (see [here](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/trainer.py#L146)), so maybe it's better to put it [here](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/reader/creator.py#L105) (in the `def reader` implementation), so every time a new pass begin, `reader()` is called, then `paddle_start_get_records` is called.

Comment (Yancey1989) [inline go/pserver/client/c/test/test_train.py]: Thanks, done.

File: python/paddle/v2/reader/creator.py [hunk 2]
```diff
@@ -98,14 +102,17 @@ def recordio(paths, buf_size=100):
     if host_name not in os.environ.keys():
```
Comment (typhoonzero) [inline python/paddle/v2/reader/creator.py]: For the line `host_name = "MASTER_SERVICE_HOST"` we don't have a "master service" currently.

File: python/paddle/v2/reader/creator.py [hunk 3]
```diff
@@ -81,6 +81,9 @@ def reader():
     return dec.buffered(reader, buf_size)
 
 
+pass_num = 0
+
+
 def recordio(paths, buf_size=100):
```
Comment (typhoonzero) [inline python/paddle/v2/reader/creator.py]: Maybe the name `recordio` as `cloud_reader` is more clear?

File: python/paddle/v2/reader/creator.py [hunk 4]
```diff
@@ -98,14 +102,17 @@ def recordio(paths, buf_size=100):
...
         raise Exception('not find ' + host_name + ' in environment variable.')
 
-    addr = os.environ(host)
+    addr = os.getenv(host_name)
 
     def reader():
         c = cloud(addr, buf_size)
         c.set_dataset(paths)
+        c.paddle_start_get_records(pass_id)
+        global pass_num
```
Comment (typhoonzero) [inline python/paddle/v2/reader/creator.py]: Put `global` the first line under function define.

File: python/paddle/v2/reader/creator.py [hunk 5]
```diff
@@ -90,6 +93,7 @@ def recordio(paths, buf_size=100):
     """
     import os
```
Comment (typhoonzero) [inline python/paddle/v2/reader/creator.py]: Need some demo code of how to use this reader.

File: python/paddle/v2/reader/creator.py [hunk 6]
```diff
@@ -90,6 +93,7 @@ def recordio(paths, buf_size=100):
...
     import paddle.v2.master.client as cloud
+    import cPickle as pickle
```
Comment (helinwang) [inline python/paddle/v2/reader/creator.py]: Do we need this, not see anywhere it being used.

File: python/paddle/v2/reader/creator.py (hunk 3 above)
Comment (Yancey1989) [inline python/paddle/v2/reader/creator.py]: Done.

File: python/paddle/v2/reader/creator.py (hunk 5 above)
Comment (Yancey1989) [inline python/paddle/v2/reader/creator.py]: Done.

File: python/paddle/v2/reader/creator.py (hunk 2 above)
Comment (Yancey1989) [inline python/paddle/v2/reader/creator.py]: Done.

File: python/paddle/v2/reader/creator.py (hunk 4 above)
Comment (Yancey1989) [inline python/paddle/v2/reader/creator.py]: Done.

File: python/paddle/v2/reader/creator.py (hunk 6 above)
Comment (Yancey1989) [inline python/paddle/v2/reader/creator.py]: Sorry, it's my mistake, we need `pickle.loads()` with the record fetched from master.
Done.

File: python/paddle/v2/reader/creator.py [hunk 7]
```diff
@@ -81,35 +81,41 @@ def reader():
...
 
-def recordio(paths, buf_size=100):
+pass_num = 0
+
+
+def cloud_reader(paths, etcd_endpoints, timeout_sec=5, buf_size=64):
     """
-    Creates a data reader that outputs record one one by one
-        from given local or cloud recordio path.
+    Create a data reader that yield a record one bye one from
```
Comment (typhoonzero) [inline python/paddle/v2/reader/creator.py]: one bye one => one by one

Comment (Yancey1989) [inline python/paddle/v2/reader/creator.py]: Done. Thanks!

**Comment by Yancey1989** [comment]
I deleted the unit test for the older `recordio` reader because it's dependencies on the etcd mater process, and we use 'https://github.com/PaddlePaddle/Paddle/blob/develop/go/pserver/client/c/test/test_train.py to do the test.

**Pull request merged**